### PR TITLE
Adopt the  kubernetes.io/role/internal-elb and kubernetes.io/role/elb tags

### PR DESCRIPTION
**Pull request by bigkraig**
This PR leverages the common Kubernetes subnet tags to auto-detect internal and internet-facing subnets.

### CODE REVIEW

File: docs/setup.md [hunk 1]
```diff
@@ -18,19 +18,24 @@ An example policy with the minimum rights can be found at [examples/iam-policy.j
...
+##### Via annotation
+`alb.ingress.kubernetes.io/subnets` may be specified in each ingress resource with the subnet IDs or `Name` tags. This allows for flexibility in where ALBs land. This list of subnets must include 2 or more that exist in unique availability zones. See the [annotations documentation](ingress-resources.md#annotations) for more details.
 
-- auto-detection: When subnet annotations are not present, the controller will attempt to choose the best subnets for deploying the ALBs. It uses the following tag criteria to determine the subnets it should use.
+##### Via tags on the subnets
+When subnet annotations are not present, the controller will attempt to choose the best subnets for deploying the ALBs. It uses the following tag criteria to determine the subnets it should use.
 
-	- `kubernetes.io/cluster/$CLUSTER_NAME`=`shared` where `$CLUSTER_NAME` matches the `CLUSTER_NAME` environment variable from the `alb-ingress-controller.yaml` manifest.
+* `kubernetes.io/cluster/$CLUSTER_NAME: = "shared"` `$CLUSTER_NAME` must match the `CLUSTER_NAME` environment variable on the controller.
```
Comment (whithajess) [inline docs/setup.md]: This should be able to be `shared` or `owned` see: https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/tags.go#L43

File: pkg/aws/albec2/ec2.go [hunk 2]
```diff
@@ -16,12 +20,21 @@ import (
...
 
+var cache = ccache.New(ccache.Configure())
+
 const (
 	instSpecifierTag = "instance"
 	ManagedByKey     = "ManagedBy"
 	ManagedByValue   = "alb-ingress"
+
+	tagNameCluster  = "kubernetes.io/cluster"
+	tagValueCluster = "shared"
```
Comment (whithajess) [inline pkg/aws/albec2/ec2.go]: Same comment as above ^

File: pkg/aws/albrgt/rgt.go [hunk 3]
```diff
@@ -43,45 +46,77 @@ func (r *RGT) GetResources(clusterName *string) (*Resources, error) {
...
-			req, _ := r.GetResourcesRequest(&params)
-			return req, nil
+		&resourcegroupstaggingapi.GetResourcesInput{
+			ResourceTypeFilters: []*string{
+				aws.String("elasticloadbalancing"),
+			},
+			TagFilters: []*resourcegroupstaggingapi.TagFilter{
+				&resourcegroupstaggingapi.TagFilter{
+					Key:    aws.String("kubernetes.io/cluster/" + *clusterName),
+					Values: []*string{aws.String("owned")},
```
Comment (whithajess) [inline pkg/aws/albrgt/rgt.go]: At here I think you want `shared` or `owned` right?

**Comment by bigkraig** [comment]
Thanks for the review @whithajess!

**Comment by argeualcantara** [comment]
Considering: 

> A subnet is public if it routes to an internet gateway and it's instances will be reachable if they are attached to an Elastic IP.

Can't we use the routing tables to determine if the subnet is public or private, the public ones have `0.0.0.0/0` set to target `igw-*`. we could use something like this to avoid the need of adding more tags to subnets:

```Go
	params := &ec2.DescribeRouteTablesInput{
		Filters: []*ec2.Filter{
			{
				Name:   aws.String("association.subnet-id"),
				Values: []*string{aws.String(subnetid)},
			},
			{
				Name:   aws.String("route.destination-cidr-block"),
				Values: []*string{aws.String("0.0.0.0/0")},
			},
			{
				Name:   aws.String("route.gateway-id"),
				Values: []*string{aws.String("igw*")},
			},
		},
	}
```

**Comment by jhohertz** [comment]
This would go a long way to allowing this ingress controller to co-exist with k8s internal LBs and other ELB/ALB ingress controllers that use this standard.

**Comment by whithajess** [comment]
fixes #253

**Pull request merged**